### PR TITLE
[WIP] Desktop: Add Unit Tests

### PR DESCRIPTION
**Pull request by rihardsgravis**
# Description

Add Unit Tests for:

- [x] React UI components
- [x] React UI views visual
- [x] Native library wrappers
- [x] SeedStore classes
- [x] End-to-end tests for:
  - [x] Onboarding existing seed 
  - [x] Onboarding new seed
  - [ ] Send & receive

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/desktop/package.json [hunk 1]
```diff
@@ -51,13 +54,18 @@
     "electron-builder": "^20.38",
     "electron-devtools-installer": "^2.2.4",
     "electron-screenshot": "^1.0.5",
+    "enzyme": "^3.8.0",
+    "enzyme-adapter-react-16": "^1.8.0",
+    "enzyme-to-json": "^3.3.5",
     "express": "^4.16.3",
     "file-loader": "^2.0.0",
     "html-webpack-plugin": "^3.2.0",
+    "jest": "^23.6.0",
```
Comment (rajivshah3) [inline src/desktop/package.json]: `jest` and `babel-jest` are bringing in vulnerable versions of `braces`, which is causing the Snyk test to fail. We need to force `braces@^2.3.1` to solve this issue. A package like https://www.npmjs.com/package/npm-force-resolutions might help, but that seems to work for lockfiles and not shrinkwraps. If I can figure out how to make it work for shrinkwraps I'll make a PR

Comment (rihardsgravis) [reply src/desktop/package.json]: You can achieve that by editing the shrinkwrap file manually and change the child dependency version there.

Comment (rajivshah3) [reply src/desktop/package.json]: Would it be overwritten by `npm install`?

Comment (rihardsgravis) [reply src/desktop/package.json]: It should not be overwritten - it's the opposite - `npm install` uses the shrinkwrap file to resolve the exact package versions (the only reason to have a shrinkwrap file at all).

Comment (rihardsgravis) [reply src/desktop/package.json]: So, jest released `24.0.0` two days ago. We are ok now 👍

**Comment by rajivshah3** [comment]
@rihardsgravis I resolved the vulnerable dependency issue in https://github.com/facebook-atom/jest-electron-runner/commit/6f3e2979d7076171d7263517b34889036e384953. Just waiting for them to create a new release so the Snyk tests will pass

**Pull request closed without merging**